cmd/ci-operator/main.go

```diff
@@ -846,10 +849,56 @@ func (o *options) writeMetadataJSON() error {
 	m.Pod = o.jobSpec.ProwJobID
 	m.WorkNamespace = o.namespace
 
-	data, err := json.MarshalIndent(m, "", "  ")
+	prowMetadataJSON, err := json.MarshalIndent(m, "", "  ")
 	if err != nil {
 		return err
 	}
+
+	// Try to find the custom prow metadata file. We assume that there's only one. If there's more than one,
+	// we'll just use the first one that we find.
+	var customProwMetadataFile string
+	filepath.Walk(o.artifactDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return filepath.SkipDir
+		}
+
+		// If the prow metadata file is set, just return, since we've already found the file
+		if customProwMetadataFile != "" {
+			return filepath.SkipDir
+		}
+
+		var fileName = info.Name()
+		if fileName == CustomProwMetadata {
+			customProwMetadataFile = path
+			return filepath.SkipDir
+		}
+
+		return nil
+	})
+
+	var data []byte
+	// Merge custom metadata into resulting JSON
+	if customProwMetadataFile != "" {
+		log.Printf("Found custom prow metadata.")
+
+		// Turns out Unmarshaling into an map is not destructive, so if we call it twice it will
+		// effectively merge these two JSON objects.
+		var mergedJSON map[string]interface{}
+		json.Unmarshal(prowMetadataJSON, &mergedJSON)
+
+		var customJSONFile []byte
+		if customJSONFile, err = ioutil.ReadFile(customProwMetadataFile); err != nil {
+			return err
```

Comment:
This would make failure to read custom JSON to not emit ci-operator's metadata. It's desirable to make the run fail, but the standard metadata needs to be emitted.